pkg/eventstream/eventstream.go

```diff
@@ -158,33 +163,65 @@ func parseMultipartStream(
 			return err
 		}
 
-		// Skip empty parts
-		if len(data) == 0 {
-			continue
-		}
-
 		fieldName := part.FormName()
 		if fieldName != "" {
-			formData[fieldName] = string(data)
+			formData[fieldName] = append(formData[fieldName], string(data))
 		}
 	}
 
-	if len(formData) > 0 {
-		byt, err := json.Marshal(formData)
-		if err != nil {
-			return err
-		}
+	byt, err := json.Marshal(formData)
+	if err != nil {
+		return err
+	}
 
-		if len(byt) > maxSize {
-			return fmt.Errorf("%w: Max %d bytes / Size %d bytes", ErrEventTooLarge, maxSize, len(byt))
-		}
+	if len(byt) > maxSize {
+		return fmt.Errorf("%w: Max %d bytes / Size %d bytes", ErrEventTooLarge, maxSize, len(byt))
+	}
 
-		select {
-		case stream <- StreamItem{N: 0, Item: json.RawMessage(byt)}:
-			// Success
-		case <-ctx.Done():
-			return nil
-		}
+	select {
+	case stream <- StreamItem{N: 0, Item: json.RawMessage(byt)}:
+		// Success
+	case <-ctx.Done():
+		return nil
+	}
+
+	return nil
+}
+
+// parseFormUrlencodedStream parses application/x-www-form-urlencoded data and
+// extracts form fields as a JSON event
+func parseFormUrlencodedStream(
+	ctx context.Context,
+	r io.Reader,
+	stream chan StreamItem,
+	maxSize int,
+) error {
+	// Read all data from the reader
+	data, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+
+	// Parse the form data
+	values, err := url.ParseQuery(string(data))
```

Comment:
cool, looks like this groups repeated fields into an array for us for free https://pkg.go.dev/net/url#ParseQuery

and I see your tests cover that too

Reply:
I don't think we should do anything to handle these but FYI some libraries have a pattern of using `field[]` or `field[0]`/`field[1]` for sending arrays in form-urlencoded

https://www.baeldung.com/java-postman-send-array
https://brandur.org/fragments/application-x-wwww-form-urlencoded

This isn't part of the spec for `application/x-www-form-urlencoded` though, so I'd leave it as is until we see some webhook in the wild that adheres to this.